implementations/rust/ockam/ockam_executor/src/channel.rs

```diff
@@ -50,28 +57,21 @@ struct Inner<T> {
     is_sender_closed: AtomicBool,
 }
 
+impl<T> Inner<T> {
+    fn _len(&self) -> usize {
```

Comment:
Nit: Both are private and underscore-prefixed, so is there a reason to have this rather than just making the caller read `_length` directly?

Reply:
Ugh, this is indeed quite awful in a number of ways:

* The fundamental mismatch here is that `ockam_executor::channel` is a drop-in replacement for [`tokio::sync::mpsc::channel`](https://docs.rs/tokio/latest/tokio/sync/mpsc/fn.channel.html)
* Tokio takes a `length` parameter for its channels and does a dynamic allocation.
* `ockam_executor::channel`, on the other hand, is a thin wrapper around the `heapless` queue which does static allocation.
* So `length` is actually never used and (🙈) on `no_std` all queues are just allocated with a fixed size of 16.

At one point I was messing around with tracking queue lengths and item counts at runtime but gave that up in favor of tackling the underlying mismatch. (Maybe replacing the heapless queue with a dynamically allocated one, maybe switching tokio::sync::mpsc::channel for a statically allocated one, maybe something else?)

Why I had a `len()` method I really can't remember! 😂

Either way, I've cut these out for now and made a short doc note about what's going on here.

Reply:
I see. Wrapping the subset of tokio's channel that we need (so that it takes `const N: usize` that it just passes to the ctor) wouldnt be too bad I suspect, since we never use actually-dynamic sizes really. If it's not causing problems now lets wait til we need it, though.

(I suspect we'll have to move some of them to unbounded eventually anyway, to avoid deadlocks like #1947. I do have another solution in a comment there for how to handle it properly, but this kind of "gridlock" deadlock is common enough that I feel like we'll hit it at some point... But maybe not on embedded)